multi-educator/src/Dscp/MultiEducator/Launcher/Mode.hs

```diff
@@ -106,47 +109,69 @@ deriveHasLens 'mecWitnessVars ''MultiEducatorContext ''W.WitnessVariables
 -- (Almost) Natural Transformation
 ----------------------------------------------------------------------------
 
+-- TODO fix comment:
 -- | This function transforms normal workmode into multi-workmode using login.
 -- It returns Nothing is login was not found
-lookupEducator :: MultiEducatorWorkMode ctx m => Text -> m EducatorCtxWithCfg
-lookupEducator login = do
+lookupEducator
+    :: MultiEducatorWorkMode ctx m
+    => EducatorAuthLogin
+    -> m EducatorCtxWithCfg
+lookupEducator educatorAuthLogin = do
     mctx <- ask
-    let MultiEducatorResources{..} = mctx ^. lensOf @MultiEducatorResources
+    let educatorId = eadId $ ealData educatorAuthLogin
+        MultiEducatorResources{..} = mctx ^. lensOf @MultiEducatorResources
+        -- Note: we need this because even if we already have the 'EducatorCtxWithCfg'
+        -- in the map it may need to be updated with a new token
+        updateLoginToken (EducatorCtxWithCfg edCtx) = do
+            certIssuerInfoRes <- makeCertIssuerRes educatorAuthLogin
+            let newEdCtx = set (E.ecResources . E.erPdfCertIssuerRes) certIssuerInfoRes edCtx
```

Comment:
Infix syntax `edCtx & ecResources.erPdfCertIssuerRes .~ certIssuerInfoRes` seems to be neater and more familiar.